### PR TITLE
Fixed VS crash during implicit conversion of null object in nullable walker

### PR DESCRIPTION
**Pull request by kevinsun-dev**
Fixed VS crash during implicit conversion due to null object. Documented in issue #45862.

Changes:
 - Added null check in Nullable Walker
 - Added test

### CODE REVIEW

**Comment by gafter** [comment]
Can you please use more descriptive titles for PRs in the future?

**Comment by kevinsun-dev** [comment]
> Can you please use more descriptive titles for PRs in the future?

Apologies, forgot to change the title from when it was a draft.

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs [hunk 1]
```diff
@@ -4780,6 +4780,37 @@ void F(System.Action<System.Object?, System.Action<System.Object!, System.Object
...
+    void M()
+    {
+        _ = 0 switch
+        {
+            0 =
+            _ = null,
+        };
+    }
+}";
+            var comp = CreateCompilation(new[] { source }, parseOptions: TestOptions.Regular8, options: WithNonNullTypesTrue(TestOptions.ReleaseModule));
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs:4801]: Can be simplified to `var comp = CreateCompilation(source);`

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs [hunk 2]
```diff
@@ -4780,6 +4780,37 @@ void F(System.Action<System.Object?, System.Action<System.Object!, System.Object
             });
         }
 
+        [Fact]
+        [WorkItem(45862, "https://github.com/dotnet/roslyn/issues/45862")]
+        public void Issue_45862()
```
[the remark this answers is not in the document]
Comment (cston) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs:4785]: Consider giving the test a descriptive name, perhaps "SwitchExpressionNoType".

---
In reply to: [455193504](https://github.com/dotnet/roslyn/pull/45974#discussion_r455193504) [](ancestors = 455193504)

**Pull request merged**